src/main/java/org/openelisglobal/common/rest/provider/PatientSearchRestController.java

```diff
@@ -138,4 +166,81 @@ private PatientSearchWorker getAppropriateWorker(HttpServletRequest request, boo
                 externalID, patientID, guid, dateOfBirth, gender);
     }
 
+    private List<org.hl7.fhir.r4.model.Patient> parseCRPatientSearchResults(Bundle patientBundle) {
+        return patientBundle.getEntry().stream()
+                .filter(entry -> entry.getResource() instanceof org.hl7.fhir.r4.model.Patient)
+                .map(entry -> (org.hl7.fhir.r4.model.Patient) entry.getResource()).collect(Collectors.toList());
+    }
+
+    private List<PatientSearchResults> searchPatientInClientRegistry(String lastName, String firstName, String STNumber,
+            String subjectNumber, String nationalID, String patientID, String guid, String dateOfBirth, String gender) {
+        if (isClientRegistryConfigInvalid()) {
+            return new ArrayList<>();
+        }
+
+        IGenericClient clientRegistry = fhirUtil.getFhirClient(fhirConfig.getClientRegistryServerUrl(),
+                fhirConfig.getClientRegistryUserName(), fhirConfig.getClientRegistryPassword());
+
+        List<PatientSearchResults> patientSearchResults = searchResultsService.getSearchResults(lastName, firstName,
+                STNumber, subjectNumber, nationalID, null, patientID, guid, dateOfBirth, gender);
+
+        List<PatientSearchResults> finalResults = new ArrayList<>();
+
+        for (PatientSearchResults patientSearchResult : patientSearchResults) {
```

Comment:
I think the Bussines Logic here isnt right. Looks like you first seravh the local DB , extract idenfiers then use the identifiers to serach from the CR. 
I think thats wrong , because if the local DB search doesnt return any result , you wouldnt serach the CR.
You could just directly do a FHIR search from the CR sing the above params ie lastName, FirstName ,etc